ispell fails to build on Darwin

## Issue description

`ispell` fails to build on Darwin with the following error message:

```
make[1]: Leaving directory '/private/tmp/nix-build-ispell-3.3.02.drv-0/ispell-3.3.02'
chmod +x tryaffix
cc -O -c term.c
term.c:95:10: fatal error: 'termio.h' file not found
#include <termio.h>
         ^~~~~~~~~~
1 error generated.
term.c wouldn't compile.  Try inverting USG in local.h
make: *** [Makefile:588: term.o] Error 1
builder for '/nix/store/kv00qwnsbwi2zl6mvi9kyivkv0d4l8i9-ispell-3.3.02.drv' failed with exit code 2
error: build of '/nix/store/kv00qwnsbwi2zl6mvi9kyivkv0d4l8i9-ispell-3.3.02.drv' failed
```

### Steps to reproduce

```bash
nix-env -i -A nixpkgs.ispell
```

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 17.6.0, macOS 10.13.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixpkgs-18.09pre133932.ee28e35ba37"`
 - channels(smaret): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

--- a/pkgs/tools/text/ispell/default.nix
+++ b/pkgs/tools/text/ispell/default.nix
@@ -22,8 +22,8 @@ stdenv.mkDerivation rec {
     ./patches/0030-Display-whole-multibyte-character.patch
   ];
   postPatch = ''
+    if [ ! stdenv.isDarwin ]; then echo "#define USG" >> local.h; fi
     cat >> local.h <<EOF
-    #define USG
     #define TERMLIB "-lncurses"
     #define LANGUAGES "{american,MASTERDICTS=american.med,HASHFILES=americanmed.hash}"
     #define MASTERHASH "americanmed.hash"

--- a/pkgs/tools/text/ispell/default.nix
+++ b/pkgs/tools/text/ispell/default.nix
@@ -22,8 +22,8 @@ stdenv.mkDerivation rec {
     ./patches/0030-Display-whole-multibyte-character.patch
   ];
   postPatch = ''
-    if [ ! stdenv.isDarwin ]; then echo "#define USG" >> local.h; fi
     cat >> local.h <<EOF
+    ${stdenv.lib.optionalString (!stdenv.isDarwin) "#define USG"}
     #define TERMLIB "-lncurses"
     #define LANGUAGES "{american,MASTERDICTS=american.med,HASHFILES=americanmed.hash}"
     #define MASTERHASH "americanmed.hash"

